ui/src/api/api.tsx

```diff
@@ -139,3 +153,23 @@ export const deleteUserRole = async (role: Role) => {
       return error.response;
     });
 }
+
+export const getIdToken = async( msalInstance: PublicClientApplication ): Promise<string> => {
+  const activeAccount = msalInstance.getActiveAccount(); // This will only return a non-null value if you have logic somewhere else that calls the setActiveAccount API
+  const accounts = msalInstance.getAllAccounts();
+  const request = {
+      scopes: ["User.Read"],
+      account: activeAccount || accounts[0]
+  };
+  await msalInstance.acquireTokenSilent(request).then(response => {
```

Comment:
>acquireTokenSilent [](http://example.com/codeflow?start=21&length=18)

It seems acquireTokenSilent will look up token from local cache first, if not found or expired, this method will fetch / refresh token. Is this correct?

If refresh token is taken be MADL internally, it's better add a comment here.